src/main/java/org/home/controller/MessageQueueController.java

```diff
@@ -0,0 +1,83 @@
+package org.home.controller;
+
+import org.apache.log4j.Logger;
+import org.home.implementation.SimpleMessageStorageImpl;
+import org.home.interfaces.MessageStorageInterface;
+import org.home.model.Message;
+import org.home.model.MessageQueue;
+import org.home.model.Owner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+
+@Component
+public class MessageQueueController {
+    List<MessageQueue> allQueue;
+
+    @Autowired
+    @Qualifier("primary")
+    MessageStorageInterface messageStorageInterface;
+    static final Logger logger = Logger.getLogger(MessageQueueController.class);
+    public MessageQueueController()
+    {
+        allQueue = new ArrayList<>();
+    }
+
+    public void addNewQueue(String qName, Owner owner) {
+        if(this.queueExists(qName)!=null)
+        {
+            logger.info(qName +" already exists");
+        }
+        else {
+            MessageQueue messageQueue = new MessageQueue(owner, qName);
+            allQueue.add(messageQueue);
+            logger.info(qName+" created");
+        }
+    }
+
+    public MessageQueue queueExists(String qName)
```

Comment:
The naming convention is not right here. 
`queueExists ` seems to be returning a boolean result, you are getting the queue object from name. 
In that case, `getQueueFromName` will be better approach